src/auto/injector.js

```diff
@@ -843,15 +832,12 @@ function createInjector(modulesToLoad, strictDi) {
         fn = fn[fn.length - 1];
       }
 
-      if (!isClass(fn)) {
+      try {
         // http://jsperf.com/angularjs-invoke-apply-vs-switch
-        // #5388
         return fn.apply(self, args);
-      } else {
-        args.unshift(null);
-        /*jshint -W058 */ // Applying a constructor without immediate parentheses is the point here.
-        return new (Function.prototype.bind.apply(fn, args));
-        /*jshint +W058 */
+      } catch (e) {
+        // if `fn` is a class, then it needs to be called using `new`
+        return new (Function.prototype.bind.apply(fn, [null].concat(args)))();
```

Comment:
It looks (from the JS specs) that it makes no difference in practice:

`new Foo` is equivalent to `new Foo()`

I wonder why @mprobst wanted to leave that out in the first place?